ooni/nettests/blocking/web_connectivity.py

```diff
@@ -164,9 +164,18 @@ def setUp(self):
         self.report['tcp_connect'] = []
         self.report['control'] = {}
 
-        self.hostname = urlparse(self.input).netloc
-        if not self.hostname:
-            raise AbsentHostname('No hostname', self.input)
+        parseResult = urlparse(self.input)
+
+        # reject --url parameter in case http(s):// was not supplied
+        if not parseResult.scheme and not parseResult.hostname:
+            if not self.input.startswith("http"): 
```

Comment:
Actually if you have entered in here (because `not parseResult.scheme == True`), it means that self.input will probably not start with `http` already (unless it has another SCHEMA).